code/modules/paperwork/paper.dm

```diff
@@ -434,6 +399,61 @@
 		return loc
 	return ..()
 
+/**
+ * ##stamp
+ *
+ * Proc used to place a stamp onto a piece of paper
+ *
+ * Arguments:
+ * * x - The x coord of the stamp
+ * * y - The y coord of the stamp
+ * * r - The rotation in degrees, of the stamp
+ * * icon_state - The stamp icon to be placed on the paper
+ * * class - (Optional) A string needed for the list of stamps on the page
+ */
+/obj/item/paper/proc/stamp(x, y, r, icon_state, class = "paper121x54 [icon_state]")
+	if (isnull(stamps))
+		stamps = list()
+	if (stamps.len < MAX_PAPER_STAMPS)
+		stamps[++stamps.len] = list(class, x, y, r)
```

Comment:
```suggestion
		stamps += list(list(class, x, y, r))
```

Reply:
Wouldn't inserting a double list break things?

Reply:
no because dm unwraps the outer list so its really stamps += list(class, x, y, r)

Reply:
I'm still not seeing the purpose of this?  If we don't need the outer list and it's getting discarded due to dm, then why include it?  Wouldn't it be better to do it as: `stamps += list(class, x, y, r)` since `stamps` is just an array.

```
stamps[1] = list(blah, blah)
stamps[2] = list(blah, blah)
stamps[3] = list(blah, blah)

VS

stamps[1] = list(list(blah, blah)) // outer list is getting discarded?
stamps[2] = list(list(blah, blah))
stamps[3] = list(list(blah, blah))
```

Reply:
#66230